src/segments/azd_test.go

```diff
@@ -0,0 +1,70 @@
+package segments
+
+import (
+	"errors"
+	"io/fs"
+	"path/filepath"
+	"testing"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/mock"
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/stretchr/testify/assert"
+	mock2 "github.com/stretchr/testify/mock"
+)
+
+func TestAzdSegment(t *testing.T) {
+	cases := []struct {
+		Case            string
+		ExpectedEnabled bool
+		ExpectedString  string
+		Template        string
+		IsInited        bool
+	}{
+		{
+			Case:            "no .azure directory found",
+			ExpectedEnabled: false,
+		},
+		{
+			Case:            "Environment located",
+			ExpectedEnabled: true,
+			ExpectedString:  "TestEnvironment",
+			Template:        "{{ .DefaultEnvironment }}",
+			IsInited:        true,
+		},
+	}
+
+	for _, tc := range cases {
+		env := new(mock.MockedEnvironment)
+		env.On("Debug", mock2.Anything)
+
+		if tc.IsInited {
+			fileInfo := &platform.FileInfo{
+				Path:         "test/.azure",
+				ParentFolder: "test",
+				IsDir:        true,
+			}
+			env.On("HasParentFilePath", ".azure").Return(fileInfo, nil)
+			dirEntries := []fs.DirEntry{
```

Comment:
instead of having `IsInited` I would expose `DirEntries` so you can also test that logic (no `config.json` for example).

Reply:
To the best of my knowledge, there's no way that you could have the `.azure` file without a `config.json` file, without having something really broken with `azd`. Would you still want to handle that scenario then?